sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/codegen/CodeGenerator.scala

```diff
@@ -56,6 +56,14 @@ class CodeGenContext {
    */
   val references: mutable.ArrayBuffer[Expression] = new mutable.ArrayBuffer[Expression]()
 
+  /**
+   * Holding expressions' mutable states like `Rand.rng`, and keep them as member variables
+   * in generated classes like `SpecificProjection`.
+   * Each element is a 3-tuple: java type, variable name, variable value.
+   */
+  val mutableStates: mutable.ArrayBuffer[(String, String, Any)] =
```

Comment:
let's add a function so we don;t call += directly on the array buffer.

```scala
def addMutableState(javaType: String, variableName: String, initialValue: String): Unit
```

this way it also makes it more clear what the semantics are for each element in the tuple3.